subprojects/core-api/src/main/java/org/gradle/api/invocation/Gradle.java

```diff
@@ -190,6 +190,24 @@
      */
     void buildStarted(Action<? super Gradle> action);
 
+    /**
+     * Adds an action to be called before the build settings have been loaded and evaluated.
+     *
+     * @param closure The action to execute.
+     * @since 5.7
+     */
+    @Incubating
+    void beforeSettings(Closure<?> closure);
```

Comment:
Isn't the `Action`-taking variant enough?

Reply:
Unfortunately not. Please see: https://gradle.slack.com/archives/C03A0MT6S/p1566182271414700

TL;DR: these closure methods behave different to most.